From 6e9281276ddd6d37040b49083d4c39b154a40eae Mon Sep 17 00:00:00 2001
From: Erickson Silva de Oliveira <Erickson.SilvadeOliveira@windriver.com>
Date: Mon, 17 Mar 2025 09:25:21 -0300
Subject: [PATCH] ceph-csi-cephfs: Add conditional to enable leader election

The rbd-provisioner and cephfs-provisioner pods use the K8s
API to conduct leadership elections, but these elections
should be disabled on AIO-SX, since they are not needed.

Signed-off-by: Erickson Silva de Oliveira <Erickson.SilvadeOliveira@windriver.com>
---
 .../ceph-csi-cephfs/templates/provisioner-deployment.yaml   | 6 ++++++
 charts/ceph-csi-cephfs/values.yaml                          | 3 +++
 2 files changed, 9 insertions(+)

diff --git a/charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml b/charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml
index 3a5287f..fe8206c 100644
--- a/charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml
+++ b/charts/ceph-csi-cephfs/templates/provisioner-deployment.yaml
@@ -139,7 +139,9 @@ spec:
             - "--csi-address=$(ADDRESS)"
             - "--v={{ .Values.sidecarLogLevel }}"
             - "--timeout={{ .Values.provisioner.timeout }}"
+{{- if .Values.provisioner.leaderElection.enabled }}
             - "--leader-election=true"
+{{- end }}
             - "--retry-interval-start=500ms"
             - "--extra-create-metadata=true"
             - "--feature-gates=HonorPVReclaimPolicy=true"
@@ -176,7 +178,9 @@ spec:
             - "--csi-address=$(ADDRESS)"
             - "--v={{ .Values.sidecarLogLevel }}"
             - "--timeout={{ .Values.provisioner.timeout }}"
+{{- if .Values.provisioner.leaderElection.enabled }}
             - "--leader-election=true"
+{{- end }}
             - "--extra-create-metadata=false"
             - "--feature-gates=CSIVolumeGroupSnapshot={{.Values.provisioner.snapshotter.args.enableVolumeGroupSnapshots }}"
 {{- if and .Values.provisioner.snapshotter.args .Values.provisioner.snapshotter.args.httpEndpointPort }}
@@ -212,7 +216,9 @@ spec:
             - "--v={{ .Values.sidecarLogLevel }}"
             - "--csi-address=$(ADDRESS)"
             - "--timeout={{ .Values.provisioner.timeout }}"
+{{- if .Values.provisioner.leaderElection.enabled }}
             - "--leader-election"
+{{- end }}
             - "--retry-interval-start=500ms"
             - "--handle-volume-inuse-error=false"
             - "--feature-gates=RecoverVolumeExpansionFailure=true"
diff --git a/charts/ceph-csi-cephfs/values.yaml b/charts/ceph-csi-cephfs/values.yaml
index 4307ce0..974a428 100644
--- a/charts/ceph-csi-cephfs/values.yaml
+++ b/charts/ceph-csi-cephfs/values.yaml
@@ -187,6 +187,9 @@ provisioner:
   # useful for deployments where the podNetwork has no access to ceph
   enableHostNetwork: false

+  leaderElection:
+    enabled: true
+
   httpMetrics:
     # Metrics only available for cephcsi/cephcsi => 1.2.0
     # Specifies whether http metrics should be exposed
--
2.34.1
